libcloud/dns/drivers/worldwidedns.py

```diff
@@ -0,0 +1,506 @@
+# limitations under the License.
+"""
+World Wide DNS Driver
+"""
+
+__all__ = [
+    'WorldWideDNSDriver'
+]
+
+from libcloud.common.types import LibcloudError
+from libcloud.common.worldwidedns import WorldWideDNSConnection
+from libcloud.dns.types import Provider, RecordType
+from libcloud.dns.types import ZoneDoesNotExistError
+from libcloud.dns.types import RecordError
+from libcloud.dns.types import RecordDoesNotExistError
+from libcloud.dns.base import DNSDriver, Zone, Record
+
+
+MAX_RECORD_ENTRIES = 40
+
+
+class WorldWideDNSError(LibcloudError):
+
+    def __repr__(self):
+        return ("<WorldWideDNSError in " +
+                repr(self.driver) +
+                " " +
+                repr(self.value) + ">")
+
+
+class WorldWideDNSDriver(DNSDriver):
+    type = Provider.WORLDWIDEDNS
+    name = 'World Wide DNS'
+    website = 'https://www.worldwidedns.net/'
+    connectionCls = WorldWideDNSConnection
+
+    RECORD_TYPE_MAP = {
+        RecordType.MX: 'MX',
+        RecordType.CNAME: 'CNAME',
+        RecordType.A: 'A',
+        RecordType.NS: 'NS',
+        RecordType.SRV: 'SRV',
+        RecordType.TXT: 'TXT',
+    }
+
+    def __init__(self, key, secret=None, reseller_id=None, secure=True,
+                 host=None, port=None, **kwargs):
+        """
+        :param    key: API key or username to used (required)
+        :type     key: ``str``
+
+        :param    secret: Secret password to be used (required)
+        :type     secret: ``str``
+
+        :param    reseller_id: Reseller ID for reseller accounts
+        :type     reseller_id: ``str``
+
+        :param    secure: Weither to use HTTPS or HTTP. Note: Some providers
+                only support HTTPS, and it is on by default.
+        :type     secure: ``bool``
+
+        :param    host: Override hostname used for connections.
+        :type     host: ``str``
+
+        :param    port: Override port used for connections.
+        :type     port: ``int``
+
+        :return: ``None``
+        """
+        super(WorldWideDNSDriver, self).__init__(key=key, secret=secret,
+                                                 secure=secure, host=host,
+                                                 port=port, **kwargs)
+        self.reseller_id = reseller_id
+
+    def list_zones(self):
+        """
+        Return a list of zones.
+
+        :return: ``list`` of :class:`Zone`
+
+        For more info, please see:
+        https://www.worldwidedns.net/dns_api_protocol_list.asp
+        or
+        https://www.worldwidedns.net/dns_api_protocol_list_reseller.asp
+        """
+        action = '/api_dns_list.asp'
+        if self.reseller_id is not None:
+            action = '/api_dns_list_reseller.asp'
+        zones = self.connection.request(action)
+        if len(zones.body) == 0:
+            return []
+        else:
+            return self._to_zones(zones.body)
+
+    def list_records(self, zone):
```

Comment:
If you implement `iterate_records` you don't need to implement `list_records` since it's already implemented in the base driver as `list(self._iterate_records(zone))`.